specs/bond-manager.md

```diff
@@ -0,0 +1,146 @@
+(base fee) does not exceed `2,000 gwei` for the entire finalization window. It would be prohibitively
+expensive to keep the settlement layer base fee this high.
+
+### Variable Bond
+
+Better bond heuristics can be used to establish a bond price that accounts for
+the time-weighted gas price. One instance of this called _Varable Bonds_ use a
+separate oracle contract, `GasPriceFluctuationTracker`, that tracks gas fluctuations
+within a pre-determined bounds. This replaces the ideal solution of tracking
+challenge costs over all L1 blocks, but provides a reasonable bounds. The initial
+actors posting this bond are responsible for funding this contract.
+
+## Contract Interface
+
+Below is a minimal interface for the bond manager contract.
+
+```solidity
+/**
+ * @title IBondManager
+ * @notice The IBondManager is an interface for a contract that handles bond management.
+ */
+interface IBondManager {
+  /**
+    * @notice Post a bond with a given id and owner.
+    * @dev This function will revert if the provided bondId is already in use.
+    * @param bondId is the id of the bond.
+    * @param owner is the address that owns the bond.
+    * @param minClaimHold is the minimum amount of time the owner must wait before reclaiming their bond.
+    */
+  function postBond(bytes32 bondId, address owner, uint64 minClaimHold) external payable;
+
+  /**
+    * @notice Seizes the bond with the given id.
+    * @dev This function will revert if there is no bond at the given id.
+    * @param bondId is the id of the bond.
+    */
+  function seizeBond(bytes32 bondId) external;
+
+  /**
+    * @notice Seizes the bond with the given id and distributes it to recipients.
+    * @dev This function will revert if there is no bond at the given id.
+    * @param bondId is the id of the bond.
+    * @param recipients is a set of addresses to split the bond amongst.
+    */
+  function seizeBondAndSplit(bytes32 bondId, address[] calldata recipients) external;
+
+  /**
+    * @notice Reclaims the bond of the bond owner.
+    * @dev This function will revert if there is no bond at the given id.
+    * @param bondId is the id of the bond.
+    */
+  function reclaimBond(bytes32 bondId) external;
+}
+```
+
+**Note**
+
+The `bytes32 bondId` can be constructed using the `keccak256` hash of a given identifier.
+For example, the `L2OutputOracle` can create a `bondId` by taking the `keccak256` hash of
+the `l2BlockNumber` associated with the output proposal since there will only ever be one
+outstanding output proposal for a given `l2BlockNumber`.
+This also avoids the issue where an output proposer can have multiple bonds if bonds were
+instead tied to the address of the output proposer.
+
+## Bond Manager Implementation
+
+Initially, the bond manager will only be used by the `L2OutputOracle` contract
+for output proposals in the attestation [dispute game](./dispute-game.md). Since
+the attestation dispute game has a permissioned set of attestors, there are no
```

Comment:
This is a bit unclear because PoPs will have a bond but for posting the output. The spec for the attestation dispute game should mention that new bonds are not posted for each step. This could mention that dispute games could either have their bond split up so that a new bond is posted to cover the next step (more capital efficient) or the bond being posted in full up front

Reply:
Tried to change it up a little bit - let me know if this is more along the lines of what you're thinking. I would note that i think it's ok to explicitly put this differentiation here so that it's clear the bond management for dispute games is different from that of the `L2OutputOracle` contract's outputs.

If we don't state this explicitly it's probably easy to confuse where bonds are posted and why.